include/CLI/App.hpp

```diff
@@ -1859,8 +1859,14 @@ class App {
             return detail::Classifier::SUBCOMMAND;
         if(detail::split_long(current, dummy1, dummy2))
             return detail::Classifier::LONG;
-        if(detail::split_short(current, dummy1, dummy2))
+        if(detail::split_short(current, dummy1, dummy2)) {
+            if(dummy1[0] >= '0' && dummy1[0] <= '9') {
+                if(get_option_no_throw(std::string{'-', dummy1[0]}) == nullptr) {
+                    return detail::Classifier::NONE;
```

Comment:
This fixes a bug mentioned earlier that -N can match when it should be a number, correct?

Reply:
yes it should fix the bug mentioned [here](https://gitter.im/CLI11gitter/Lobby?at=5d5090521dadc42a11535752)